services/server/src/wss/handlers/connection.ts

```diff
@@ -25,9 +25,15 @@ export class Connection extends WssHandler {
 
       await cache.onlineStatus.addUser(user.id);
       this.announceStatus({ userId: user.id, online: true });
-
-      await this.joinRooms();
-      if (isAdmin(this.user)) this.emitServerStats();
+      this.joinChatRooms();
+
+      if (isStudent(this.user)) {
+        this.socket.join(Wss.Room.TutorsCache);
+      }
+      else if (isAdmin(this.user)) {
+        this.socket.join(Wss.Room.ServerStats);
+        this.emitServerStats();
+      }
```

Comment:
It is better to use list of `if` statements instead of `if/else`. 

```suggestion
      if (isStudent(this.user)) {
        this.socket.join(Wss.Room.TutorsCache);
      }
      
      if (isAdmin(this.user)) {
        this.socket.join(Wss.Room.ServerStats);
        this.emitServerStats();
      }
```

Reply:
`if/else` is more efficient. Can the admin be a student?
I know this is just _slightly_ efficient... but I am putting in my mind that thousands of users maybe using this function daily so I'm doing the best I can.

Reply:
> Can the admin be a student?

It will never happen.

Reply:
Then we may use `if/else if`.